# **Repository Description**

This repository complements the manuscript "Computational modelling reveals and predicts flux differences in nucleotide metabolism in IL-4-treated Cyp27a1-KO bone marrow-derived macrophages". It contains metabolic models, scripts, spreadsheets and a tutorial on how to perform a gene knockout of interest, as well as performing downstream processing and vizualization schemes. The work has been performed on Macintosh M1 Pro 2021 machine running OSX Sonoma 14.5. 

## Major software dependencies:

1) Sequence-level analysis:

STAR v2.7.11a (source: https://github.com/alexdobin/STAR) <br />
Salmon v1.10.1 (source: https://combine-lab.github.io/salmon/getting_started/) <br />
FastQC v0.12.1 (source: https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) <br />
Multiqc v1.16 (source: https://github.com/MultiQC/MultiQC) <br />

2) In MATLAB environment:

MATLAB v2023a <br />
RAVEN toolbox v2.8.4 (installation source: https://github.com/SysBioChalmers/RAVEN; includes core software resource: Task-driven Integrative Network Inference for Tissues, tINIT) <br />
COBRA toolbox v3.4 (installation guide: https://opencobra.github.io/cobratoolbox/stable/installation.html) <br />
Mouse1 Metabolic Model v1.7.0 (accessible from https://github.com/SysBioChalmers/Mouse-GEM) <br />
Gurobi - optimization suite v9.5.2 (installation guide with free academic version: https://www.gurobi.com/academia/academic-program-and-licenses/) <br />
IBM CPLEX v12.10 Optimization Studio (free academic version: https://www.ibm.com/support/pages/ibm-ilog-optimization-academic-initiative) <br />



3) In R environment:

R v4.3.1 <br />
DESeq2 v1.40.2 <br />
tximeta v1.18.1 <br />
clusterProfiler v4.8.2 <br />

 
