**Repository Description**

This repository complements the manuscript "Computational modelling reveals and predicts flux differences in nucleotide metabolism in IL-4-treated Cyp27a1-KO bone marrow-derived macrophages". It contains metabolic models, scripts, spreadsheets and a tutorial on how to perform a gene knockout of interest, as well as performing downstream processing and vizualization schemes. 

Major software dependencies:

1) Sequence-level analysis:

  STAR v2.7.11a
  Salmon v1.10.1
  FastQC v0.12.1
  Multiqc v1.16

2) In MATLAB environment:

MATLAB v2023a
RAVEN toolbox v2.8.4 (includes core software resource: Task-driven Integrative Network Inference for Tissues, tINIT)
COBRA toolbox v3.4
Mouse1 Metabolic Model v1.7.0 (accessible from https://github.com/SysBioChalmers/Mouse-GEM)
Gurobi - optimization suite v9.5.2
IBM CPLEX v12.10 Optimization Studio



3) In R environment:

R v4.3.1
DESeq2 v1.40.2
tximeta v1.18.1
clusterProfiler v4.8.2

 
